internal/controllers/core/portforward/reconciler.go

```diff
@@ -16,88 +21,75 @@ import (
 )
 
 type Reconciler struct {
-	kClient k8s.Client
+	store      store.RStore
+	kClient    k8s.Client
+	ctrlClient ctrlclient.Client
 
 	// map of PortForward object name --> running forward(s)
-	activeForwards map[string]portForwardEntry
+	activeForwards map[types.NamespacedName]portForwardEntry
 }
 
-func NewReconciler(kClient k8s.Client) *Reconciler {
+var _ store.TearDowner = &Reconciler{}
+var _ reconcile.Reconciler = &Reconciler{}
+
+func NewReconciler(store store.RStore, kClient k8s.Client, ctrlClient ctrlclient.Client) *Reconciler {
```

Comment:
You won't be able to take `ctrlclient.Client` as a constructor param, you have to let it be "injected" via `SetClient()`. (Your `Reconcile()` method won't be called _before_ that though, so you don't have to worry about it being nil.)

There's a weird chicken-and-egg party going on between the reconcilers + the controller manager (manager creates client but not until it's actually initialized/running which is _after_ the clients get constructed), see #4250 for how we ended up here.